src/doc/online_decoding.dox

```diff
@@ -313,7 +313,63 @@ phones as appropriate, e.g. <code>--ivector-silence-weighting.silence-phones=1:2
 (this should be a list of silence or noise phones in your phones.txt; you can experiment with
 which ones to include).
 
+\subsection online_decoding_nnet2_lm Example for using your own language model with existing online-nnet2 models
+Oftentimes users will have to use their own language model to improve the
+recognition accuracy. In this section we will explain how to build a language
+model with SRILM, and how to incorporate this language model to the existing
+online-nnet2 models.
+
+We first have to build an ARPA format language model with SRILM. Note that SRILM
+comes with a lot of training options, and we assume it's the user's
+responsibility to figure out what is the best setting for their own application.
+Suppose "train.txt" is our language model training corpus (e.g., training
+data transcriptions), and "wordlist" is our vocabulary (e.g., recognizer's
+vocabulary), we can build a 3gram Kneser-Ney language model using the following
+SRILM command
```

Comment:
Yes that makes sense. i'll add one more section. will have to be offline
for couple of hours, i'll do when i get back.

Guoguo
On Oct 4, 2015 3:20 PM, "Daniel Povey" <notifications@github.com> wrote:

> In src/doc/online_decoding.dox
> <https://github.com/kaldi-asr/kaldi/pull/203#discussion_r41101233>:
>
> > @@ -313,7 +313,63 @@ phones as appropriate, e.g. <code>--ivector-silence-weighting.silence-phones=1:2
> >  (this should be a list of silence or noise phones in your phones.txt; you can experiment with
> >  which ones to include).
> >
> > +\subsection online_decoding_nnet2_lm Example for using your own language model with existing online-nnet2 models
> > +Oftentimes users will have to use their own language model to improve the
> > +recognition accuracy. In this section we will explain how to build a language
> > +model with SRILM, and how to incorporate this language model to the existing
> > +online-nnet2 models.
> > +
> > +We first have to build an ARPA format language model with SRILM. Note that SRILM
> > +comes with a lot of training options, and we assume it's the user's
> > +responsibility to figure out what is the best setting for their own application.
> > +Suppose "train.txt" is our language model training corpus (e.g., training
> > +data transcriptions), and "wordlist" is our vocabulary (e.g., recognizer's
> > +vocabulary), we can build a 3gram Kneser-Ney language model using the following
> > +SRILM command
>
> Guoguo, I think we need to cover the case where the vocabulary might be
> different, as this will almost always be the case. Of course this means
> preparing the dict directory as a modified version of the dict directory
> from the original build, in a compatible way (at least explain what this
> means, and maybe mention g2p sequitur without giving further details at
> this point)-- and then calling prepare_lang.sh with the
> --phone-symbol-table option set to the old directory's phones.txt to make
> sure it's compatible.
> Actually it might make sense to put this in a separate subsection.
> Please be more explicit that 'wordlist' is all the words in words.txt
> (except, I guess and #0)... I guess at this point you could point down to
> the separate subsection talking about the case where it's not that exact
> list of words.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/203/files#r41101233>.
>